pkg/controller/controller.go

```diff
@@ -387,7 +387,7 @@ func AttachmentPods(ownerPod *k8sv1.Pod, podInformer cache.SharedIndexInformer)
 	return attachmentPods, nil
 }
 
-func ApplyNetworkInterfaceRequestOnVMISpec(vmiSpec *v1.VirtualMachineInstanceSpec, request *v1.VirtualMachineInterfaceRequest) *v1.VirtualMachineInstanceSpec {
+func ApplyNetworkInterfaceRequestOnVMTemplateSpec(vmiSpec *v1.VirtualMachineInstanceSpec, request *v1.VirtualMachineInterfaceRequest) *v1.VirtualMachineInstanceSpec {
```

Comment:
nit: The new name would make sense if it would receive VM template spec object, but since it recivce VM-spec object nothing prevents from it to process VMI spec and be missused.